x-pack/plugins/observability_solution/inventory/public/components/entities_grid/entity_name/index.tsx

```diff
@@ -29,7 +28,7 @@ export function EntityName({ entity }: EntityNameProps) {
   const handleLinkClick = useCallback(() => {
     telemetry.reportEntityViewClicked({
       view_type: 'detail',
-      entity_type: entity['entity.type'],
+      entity_type: entity.entity.type,
```

Comment:
I'm not a huge fan of `entity.entity.<something>`.  In a follow up PR, the APIs should return interfaces decoupled from the index mapping.

Reply:
Decided to do this in this PR